### PR TITLE
Drop node v18

### PR DESCRIPTION
**Pull request by mcollina**

### CODE REVIEW

**Comment by mcollina** [comment]
cc @trivikr

**Comment by KhafraDev** [comment]
I haven't seen any reason to drop v18. Nothing in fetch or other web specs needs it gone, and it's in maintenance mode for another 5 months. Is there any reason other than precedence? If precedence is the main concern, there were genuine issues with previous versions of node that required us to drop them - I can't think of any similar issues in v18.

**Comment by trivikr** [comment]
@KhafraDev I'd provided estimates on how long unidici maintainers have to support Node.js 18.x in https://github.com/nodejs/undici/issues/3386#issuecomment-2489199082 after it's end-of-life.
In summary, it needs to be supported for at least 6 months under the assumption that undici@v8 is released in Dec 2025.

Matteo also provided how dropping Node.js 18.x will ease maintenance of undici@v7 in  https://github.com/nodejs/undici/issues/3386#issuecomment-2498533727

**Review by trivikr — APPROVED** [review]
LGTM, but this needs approval from a maintainer.

**Review by ronag — APPROVED** [review]

File: package.json
Comment (RobinTail) [inline package.json:131]: what was the reason for this particular version?

Comment (mcollina) [reply package.json:131]: Because it's the latest version that was available at the time of the release. You should not be using _anything else_.

Comment (mcollina) [reply package.json:131]: > The latest Android version is 15, so the all developers should do applications only for Android 15?

This is not a good analogy. A good analogy is slightly different: should Google recommend the use of _anything_ but the latest patch available for that given version? As we speak, this is 15.0.0_r5. Why would Google recommend the use of a version that contains known vulnerabilities? You can find the details related to Android CVEs here: https://source.android.com/docs/security/bulletin/2024-11-01#spl-vulnerability-details.

Comment (AlttiRi) [reply package.json:131]: [`engines.node`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#engines) field is not about "recommendations". 

The purpose of this field is to show which versions of Node.js the library will work on and which will not.

Wrong use of this field feels like not a recommendation, but a force to update Node.js.

**Pull request merged**
